### ISSUE TITLE
Mounting a specific partition on the boot device

### ISSUE THREAD

**Issue by GalGreenfield** [microsoft/WSL#6015]
#### Is your feature request related to a problem? Please describe.**
I read in the docs that this is currenlty not supported. Do you have plans to add this support? I currently want to add a Linux partition to be used by WSL on the same boot device (disk) that my Windows is installed on.

#### Describe the solution you'd like
For this feature to be implemented into `wsl --mount`.

#### Describe alternatives you've considered
I don't think there are any. Only other thing I can do is get another drive, boot to the other partition (Linux) copy the contents of that partition to the other drive, and mount that drive - that's not what I'm looking to do - I want to use the single drive I have.

#### Additional context
It'd enable having multiple Linux distros and generally multiple Linux-supported partitions that aren't supported by Windows to be used by WSL.

I moved back to Windows due to lack of support of games and some softwares in Linux, and now I'm a bit bummed out that I even thoughWSL exists, I can't access all of my files and existing distro installation, and either have to start from scartch, boot to it separately, or not use it at all, making WSL useless for me for this use case.

**Comment by GalGreenfield** [comment]
Oh, thanks, I didn't see that other one.

**Comment by GalGreenfield** [comment]
> /dupe #5997

It's not exactly the same situation - I'm talking about a partition in the same drive - that question talks about mounting a different drive.

**Comment by therealkenc** [comment]
Okay, will bifurcate as more expedient than getting into the underlying cause of the limitations.

**Comment by eteq** [comment]
While I think I understand from #5997 that this is a limitation of HyperV, I think the motivation here still stands.  And to add a bit more to the motivation @calgreenfield gave: in my case this is *critical* because I am on a laptop with only room for a single physical drive.  Thus it's *impossible* for me to use this mount command in any useful way in a dual-boot configuration.  In practice this has driven me away from WSL and to booting into Linux because I cannot have a two-way "share" of information between WSL and the native linux.  So for me this is probably the single biggest impediment to WSL use now.

**Comment by OneBlue** [comment]
Thanks for creating this @GalGreenfield. 

Attaching single partitions to WSL2 is a scenario that is absolutely on our radar, but as you can imagine, this requires a lot of underlying infrastructure change so it's not something that will happen soon, but we definitely want to support this at some point.

**Comment by guillermo** [comment]
What a pity. Just spent the evening installing Windows Insider to use this functionality, as indicated by the documentation offered here: [tutorial](https://docs.microsoft.com/en-us/windows/wsl/wsl2-mount-disk).

I think this "little" caveat is important enough to mention in the documentation.

**Comment by OneBlue** [comment]
@guillermo: The is specified under the `Limitations` section of the documentation link you shared.

```
At this time, only entire disks can be attached to WSL 2, meaning that it's not possible to attach only a partition. Concretely, this means that it's not possible to use wsl --mount to read a partition on the boot device, because that device can't be detached from Windows.
```

**Comment by maximilianoteruel** [comment]
I think this would be an extremely useful feature, because you will be able to dual boot (on a single disk) and read the Linux partition from Windows.

I understand that it is not a simple feature to implement, but does anyone know if there is a plan to implement it in W11 (October release)?

**Comment by maximilianoteruel** [comment]
don't let this post die...

**Comment by maximilianoteruel** [comment]
don't let this post die...

**Comment by maximilianoteruel** [comment]
Hi @OneBlue .. Do you know if this feature has any progress?

**Comment by alou-S** [comment]
My use for this feature would be to mount my encrypted Linux boot partition on Windows. Would really love to see this feature be implemented soon.

**Comment by papaj-na-wrotkach** [comment]
Any progress on this feateure?

**Comment by patroza** [comment]
I would like to use a second disk a portion for WSL direct attach, and another portion for Windows to use, so this would come in handy!
Right now im wasting a 2TB SSD for just 100GB of space required under WSL, only to benefit from IO performance and CPU usage reduction :)

**Comment by ProgramFan** [comment]
I need this feature to open luks-encrypted linux partition.

**Comment by Korazza** [comment]
Yes, I need this feature too. I would use Windows more often fore sure.

**Comment by peigongdsd** [comment]
Wondering will this feature ever get implemented?

**Comment by xCirno1** [comment]
Need this too, I don't want to keep mounting and unmounting.

**Comment by ChenPi11** [comment]
https://github.com/ChenPi11/WSL/actions/runs/30080380085
I'm trying use NBD + vsock to mount Linux partition on boot drive.
If you `wsl --mount \\.\PhysicalDrive0` --partition 3, it will mirror partition 3 to a block device on `/dev/nbd0`.
But NBD doesn't support vsock, so `/init` must forwarding AF_VSOCK to AF_UNIX in userspace.
Also, Hyper-V's implement of vsock has been reported unstability in high payload.
It increased unstability, so this solution is **read-only** and experimential, otherwise, OOM killer + NBD + VSOCK + Btrfs ~+ vibe coding~ will break your data.

**Comment by ChenPi11** [comment]
If we can build a stable brige between ntoskrnl and vmlinux, this solution will be stable to read-write. Current unstability is from vsock and userspace socket forwarding.
